jans-client-api/server/src/main/java/io/jans/ca/server/rest/OAuth20Resource.java

```diff
@@ -0,0 +1,127 @@
+package io.jans.ca.server.rest;
+
+import io.jans.ca.common.CommandType;
+import io.jans.ca.common.params.*;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+@Path("/")
+public class OAuth20Resource extends BaseResource {
+
+    @POST
+    @Path("/register-site")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response registerSite(String params) {
+        logger.info("Api Resource: /register-site  Params: {}", params);
+        String result = process(CommandType.REGISTER_SITE, params, RegisterSiteParams.class, null, null);
+        return Response.ok(result).build();
+    }
+
+    @POST
+    @Path("/update-site")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateSite(@HeaderParam("Authorization") String authorization, @HeaderParam("AuthorizationRpId") String authorizationRpId, String params) {
+        logger.info("Api Resource: /update-site  Params: {}", params);
+        String result = process(CommandType.UPDATE_SITE, params, UpdateSiteParams.class, authorization, authorizationRpId);
+        return Response.ok(result).build();
+    }
+
+    @POST
+    @Path("/remove-site")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response removeSite(@HeaderParam("Authorization") String authorization, @HeaderParam("AuthorizationRpId") String authorizationRpId, String params) {
+        logger.info("Api Resource: /remove-site  Params: {}", params);
+        String result = process(CommandType.REMOVE_SITE, params, RemoveSiteParams.class, authorization, authorizationRpId);
+        return Response.ok(result).build();
+    }
+
+    @POST
+    @Path("/get-client-token")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getClientToken(String params) {
+        logger.info("Api Resource: /get-client-token  Params: {}", params);
+        String result = process(CommandType.GET_CLIENT_TOKEN, params, GetClientTokenParams.class, null, null);
+        return Response.ok(result).build();
+    }
+
+    @POST
+    @Path("/get-access-token-by-refresh-token")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response getAccessTokenByRefreshToken(@HeaderParam("Authorization") String authorization, @HeaderParam("AuthorizationRpId") String authorizationRpId, String params) {
+        logger.info("Api Resource: /get-access-token-by-refresh-token  Params: {}", params);
+        String result = process(CommandType.GET_ACCESS_TOKEN_BY_REFRESH_TOKEN, params, GetAccessTokenByRefreshTokenParams.class, authorization, authorizationRpId);
```

Comment:
Scheduled improvement to has each operation as injectable
https://github.com/JanssenProject/jans/issues/1532